### PR TITLE
[WIP] Add AD infrastructure support

### PR DESCRIPTION
**Pull request by f-trivino**

### CODE REVIEW

File: ansible/roles/runner/tasks/setup.yml [hunk 1]
```diff
@@ -22,23 +22,51 @@
   with_items:
```
Comment (f-trivino) [inline ansible/roles/runner/tasks/setup.yml]: - NetworkManager is still missing here!.

File: ansible/roles/runner/tasks/setup.yml [hunk 2]
```diff
@@ -22,23 +22,52 @@
...
+    - python3-winrm
+    - python2-winrm
+    - NetworkManager
+    - python-dnf
     - redhat-rpm-config
     - crontabs
   notify:
     - restart_nfs
 
+# FIXME: something is breaking python3-six installation,
```
Comment (f-trivino) [inline ansible/roles/runner/tasks/setup.yml]: @netoarmando, I have tested this and it doesn't work:

task was executed: 
TASK [runner : force reinstall python3-six] ******************************************************************************************
changed: [10.8.247.3]

but:
Feb 05 11:02:48 ad bash[28623]:     import urllib3
Feb 05 11:02:48 ad bash[28623]:   File "/usr/lib/python3.6/site-packages/urllib3/__init__.py", line 8, in <module>
Feb 05 11:02:48 ad bash[28623]:     from .connectionpool import (
Feb 05 11:02:48 ad bash[28623]:   File "/usr/lib/python3.6/site-packages/urllib3/connectionpool.py", line 11, in <module>
Feb 05 11:02:48 ad bash[28623]:     from .exceptions import (
Feb 05 11:02:48 ad bash[28623]:   File "/usr/lib/python3.6/site-packages/urllib3/exceptions.py", line 2, in <module>
Feb 05 11:02:48 ad bash[28623]:     from .packages.six.moves.http_client import (
Feb 05 11:02:48 ad bash[28623]: ModuleNotFoundError: No module named 'urllib3.packages.six'
Feb 05 11:02:48 ad systemd[1]: prci.service: Main process exited, code=exited, status=1/FAILURE
Feb 05 11:02:48 ad systemd[1]: prci.service: Failed with result 'exit-code'.

Surprisingly, I executed the command by hand and worksforme... so I think we should move this tasks to the end of the playbook.

Comment (f-trivino) [reply ansible/roles/runner/tasks/setup.yml]: Moving this WA to deploy_pr_ci.yml playbook is the solution. This step is now validated:

ad                : ok=42   changed=13   unreachable=0    failed=0

systemctl status prci
Active: active (running)

File: tasks/vagrant.py [hunk 3]
```diff
@@ -26,6 +26,100 @@ def wrapper(self, *args, **kwargs):
...
+        task.execute_subtask(VagrantUpADRoot(timeout=None))
+        task.execute_subtask(VagrantUpADChild(timeout=None))
+
+    def prepare_ansible_config():
+        cfg_path = '.vagrant/provisioners/ansible/inventory/vagrant_ansible_inventory'
+        task.execute_subtask(PopenTask(['mkdir', '-p', '.vagrant/provisioners/ansible/inventory/']))
+        task.execute_subtask(PopenTask(['touch', cfg_path]))
+
+    def cfg_for_windows():
+        cfg_path = '.vagrant/provisioners/ansible/inventory/vagrant_ansible_inventory'
```
Comment (netoarmando) [inline tasks/vagrant.py]: I don't have this file. Is it created by running vagrant manually? Is it just the IP address of the `ad` host?

**Comment by netoarmando** [comment]
To do:
- Update `ansible/roles/machine/provision/templates/ipa-test-config.yaml` to support AD domains
- Apparently provisioning and tests are being executed N times, where N seems to be the number of hosts on the topology
- Windows images are not available publicly

Sample results can be found here: http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/758e5a58-2e8a-11e9-af64-52540058ccf4/

**Comment by netoarmando** [comment]
Closing this in favor of #253.

**Pull request closed without merging**